### ISSUE TITLE
`merge_cubes` returns a `ValueError` when not specifying an `overlap_resolver`

### ISSUE THREAD

**Issue by JanssenBrm** [Open-EO/openeo-python-client#522]
There seems to be an issue or inconsistency with the `merge_cubes` process logic that is implemented in the Python client library. The current version of the client sets a default value of `UNSET` for the `overlap_resolver` parameter  (https://github.com/Open-EO/openeo-python-client/blob/master/openeo/processes.py#L1927).  Also the documentation states that the `overlap_resolver` is an optional parameter (https://open-eo.github.io/openeo-python-client/api-processes.html#openeo.processes.merge_cubes). 

However, not providing a valid `overlap_resolver` seems to result in an error. It looks like it's caused by the `build_child_callback` which is called to build the callback function. In this function, it checks the type of the `process` parameter (https://github.com/Open-EO/openeo-python-client/blob/master/openeo/rest/_datacube.py#L273). For the `merge_cubes` process, this is set to the `overlap_resolver`, which is `UNSET`.  This seems to trigger a `ValueError` as the type of the `overlap_resolver` does not match with one of the checks in the `build_child_callback` function.

Some example code to reproduce the issue:

```python
import openeo
import json

from openeo.processes import merge_cubes

connection = openeo.connect(
    "openeo.dataspace.copernicus.eu"
).authenticate_oidc()

polygon = {
    "coordinates": [
        [
            [
                5.179169745059369,
                51.24984286550534
            ],
            [
                5.170016107999743,
                51.25052999567865
            ],
            [
                5.171081610725707,
                51.24861004739975
            ],
            [
                5.178604705735125,
                51.246720335821465
            ],
            [
                5.179169745059369,
                51.24984286550534
            ]
        ]
    ],
    "type": "Polygon"
}
temporal_extent =  ['2023-01-01', '2023-01-30']


# Building one datacube
base_s2 = connection.load_collection('SENTINEL2_L2A',
                                     spatial_extent=polygon,
                                     temporal_extent=temporal_extent,
                                     bands=["B04", "B08", "SCL"])
base_s2 = base_s2.process("mask_scl_dilation", data=base_s2, scl_band_name="SCL")
ndvi = base_s2.ndvi(red="B04", nir="B08", target_band='NDVI')
ndvi = ndvi.filter_bands(bands=['NDVI'])

# Building a second datacube
s1_grd = connection.load_collection('SENTINEL1_GRD',
                                    spatial_extent=polygon,
                                    temporal_extent=temporal_extent,
                                    bands=['VV'])

# Merge them without an overlap resolver
merged = merge_cubes(ndvi, s1_grd)



```

**Comment by JanssenBrm** [comment]
It looks like this change was introduced since version `0.23.0`

**Comment by soxofaan** [comment]
current workaround is to use the `DataCube.merge_cubes` method instead of the `openeo.processes.merge_cubes` function I think. Something like
```
merged = ndvi.merge_cubes(s1_grd)
```


but indeed good catch, both approaches should work

**Comment by soxofaan** [comment]
and just for completeness, just passing `None` does not work as workaround either

```
merged = merge_cubes(ndvi, s1_grd, overlap_resolver=None)
```
gives
```
File ~/src/openeo/openeo-python-client/openeo/rest/_datacube.py:319, in build_child_callback(process, parent_parameters, connection)
    317     pg = process["process_graph"]
    318 else:
--> 319     raise ValueError(process)
    321 return PGNode.to_process_graph_argument(pg)

ValueError: None
```

**Comment by soxofaan** [comment]
fixed by [e00033b](https://github.com/Open-EO/openeo-python-client/commit/e00033b10bd9beddcdc850ae9bc3c7f3003b253f)
